### ISSUE TITLE
Relation Section

### ISSUE THREAD

**Issue by philsturgeon** [apiaryio/api-blueprint#256]
I think the relation section needs expanding, as I have no idea what it means and the examples don't really add anything. 

https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#relation-section

Could anyone knock out some notes? Then I could experiment and write it up properly.

**Comment by zdne** [comment]
@philsturgeon I'll take it. As said in the spec, it is used for setting the 

>  link relation type for the given action as specified by RFC 5988.

For example `edit` 

>  o  Relation Name: edit
>   o  Description: Refers to a resource that can be used to edit the
>      link's context.
>   o  Reference: [RFC5023]
> https://www.ietf.org/rfc/rfc5023

Denotes an action (on a target resource) that has semantical meaning of `that can be used to edit the link's (= source resource) content. 

Given: 

```apib
# Article [/article]
- attributes
    - text (string)
    
## List [GET]
- relation: self

- repsonse 200 (application/json)
    - attributes (Article)
   
## Edit [PATCH]
- relation: edit

- request (application/json)
    - attributes (Article)

- repsonse 200 (application/json)
    - attributes (Article)
```

Does this clarify? This is needed for fully hypermedia systems where you define and rely only on the semantics of relations and data. So you can build universal clients regardless of domain or instance of domain. 

Check this concept if you are interest in more: https://github.com/resource-blueprint/resource-blueprint#gist-resource 

Bottom line: The relation type is supposed to be an "id" unlike action name with is just a human readable label.

**Comment by philsturgeon** [comment]
Gotcha. While I get the concept of naming a link to give it a computer recognizable identifier, I assume links need to be referenced somehow and the original example does not show this, neither does yours.

It would be good to see the list items referencing the edit link or something, so I can see how it actually fits together. 

Does that make sense?

**Comment by zdne** [comment]
Yup it does. 

At this moment the relation type is purely informational albeit some generic clients – such as Hyperdrive – are already using it. 

Check [this](https://github.com/the-hypermedia-project/Hyperdrive#transitioning-from-one-state-to-another) part of Hyperdrive's README that hopefully explains how the relation type can(should) be utilized. 

The relation type is not necessarily provided for linking within a blueprint document. It is however the integral part of an API domain (semantics) as I am hinting on http://goodapi.design (describe what you can do with the data).

In hypermedia media types such as HAL or Siren relation types maps to 

> A relation aka. 'rel' (the name of the link) 
> https://github.com/mikekelly/hal_specification/blob/master/hal_specification.md#links

or 

> Defines the relationship of the link to its entity, per Web Linking (RFC5988). 
> https://github.com/kevinswiber/siren#rel-1

**Comment by zdne** [comment]
With that being said, I plan to blog™ on this topic. Open question is what would be the best way to share this knowledge in API Blueprint specification / examples without making even more heavy...

**Comment by zdne** [comment]
We have added additional explanation on this in https://github.com/apiaryio/api-blueprint/blob/master/Advanced%20Tutorial.md#relation-types

**Comment by philsturgeon** [comment]
It’s still only half of an example surely?

Just showing that a keyword can be used, but not showing what it looks like or how other things reference it or anything else connected to it seems a bit confusing. Basically, I’m confused :)
